src/main/java/com/customer/service/CustomerServiceImpl.java

```diff
@@ -9,69 +10,59 @@
 
+	private CustomerRepository customerRepository;
+
 	@Autowired
-	CustomerRepository customerRepo;
+	CustomerServiceImpl(CustomerRepository customerRepository) {
+		this.customerRepository = customerRepository;
+	}
+
 	@Value("${spring.customer.expDate}")
-	private int expdate;
+	private int expiryDate;
 
 	@Override
-	public void createCustomer(Customer cust) {
-		cust.setExpiryDate(expdate);
-		customerRepo.save(cust);
+	public Customer createCustomer(Customer customer) {
+		customer.setExpiryDate(expiryDate);
+		return customerRepository.save(customer);
 	}
 
 	@Override
 	public List<CustomerResponse> getAllCustomers() {
-		List<Customer> cust = customerRepo.findAll();
-		List<CustomerResponse> custResp = cust.stream().map(cus -> {
-			CustomerResponse customerResponse = CustomerResponse.builder().id(cus.getId()).firstName(cus.getAddress())
-					.lastName(cus.getLastName()).address(cus.getAddress()).joiningDate(cus.getJoiningDate())
-					.expiryDate(cus.getExpiryDate()).build();
+		List<Customer> customerList = customerRepository.findAll();
+		List<CustomerResponse> customerResponseList = customerList.stream().map(cus -> {
+			CustomerResponse customerResponse = CustomerResponse.builder().id(cus.getId()).firstName(cus.getFirstName())
+					.lastName(cus.getLastName()).joiningDate(cus.getJoiningDate()).expiryDate(cus.getExpiryDate())
+					.address(cus.getAddress())
+					.build();
 			customerResponse.setStatus(cus.getJoiningDate(), cus.getExpiryDate());
 			return customerResponse;
 		}).collect(Collectors.toList());
-		return custResp;
-	}
-
-	@Override
-	public Optional<Customer> findCustomerById(int id) {
-		// TODO Auto-generated method stub
-		return null;
-	}
-
-	@Override
-	public void deleteCustomerById(int id) {
-		// TODO Auto-generated method stub
-
-	}
-
-	@Override
-	public void updateCustomer(Customer cust) {
-		// TODO Auto-generated method stub
-
-	}
-
-	@Override
-	public void deleteAllCustomers() {
-		// TODO Auto-generated method stub
-
+		Collections.sort(customerResponseList, new SortingComparator());
+		return customerResponseList;
 	}
 
 	@Override
-	public Optional<Customer> findProductById(int id) {
+	public CustomerResponse findCustomerById(int id) {
 		// TODO Auto-generated method stub
-		return null;
-	}
-
-	@Override
-	public void createProduct(Customer cust) {
-		customerRepo.save(cust);
+		Optional<Customer> customer = customerRepository.findById(id);
+		if (customer.isPresent()) {
+			CustomerResponse customerResponse = CustomerResponse.builder().id(customer.get().getId())
+					.firstName(customer.get().getFirstName()).lastName(customer.get().getLastName())
+					.joiningDate(customer.get().getJoiningDate()).expiryDate(customer.get().getExpiryDate())
+					.address(customer.get().getAddress()).build();
+			customerResponse.setStatus(customer.get().getJoiningDate(), customer.get().getExpiryDate());
+			return customerResponse;
+		}
+		else
+			throw new CustomerNotFoundException("");
```

Comment:
There should be some error message here.

Reply:
Where is @ControllerAdvise being used ? I dont see the definition anywhere

Reply:
done , was not commited.